core/src/main/java/org/apache/iceberg/rest/auth/AuthManagers.java

```diff
@@ -42,6 +57,9 @@ public static AuthManager loadAuthManager(String name, Map<String, String> prope
       case AuthProperties.AUTH_TYPE_BASIC:
         impl = AuthProperties.AUTH_MANAGER_IMPL_BASIC;
         break;
+      case AuthProperties.AUTH_TYPE_OAUTH2:
+        impl = AuthProperties.AUTH_MANAGER_IMPL_OAUTH2;
+        break;
       default:
         impl = authType;
```

Comment:
can we also define something like `AUTH_IMPL = "rest.auth.type";` that could be used here for any custom auth manager/provider we want to bring in?

Reply:
That's precisely the intent here; if you have a custom manager, you would activate it with `rest.auth.type=my.custom.AuthManager`. Is that OK for you?